intrastat_product/account_invoice.py

```diff
@@ -0,0 +1,88 @@
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.model
+    def _default_intrastat_transaction(self):
+        """ placeholder for localisation modules """
+        return self.env['intrastat.transaction'].browse([])
+
+    intrastat_transaction_id = fields.Many2one(
+        'intrastat.transaction', string='Intrastat Transaction Type',
+        default=_default_intrastat_transaction, ondelete='restrict',
+        help="Intrastat nature of transaction")
+    intrastat_transport_id = fields.Many2one(
+        'intrastat.transport_mode', string='Intrastat Transport Mode',
+        ondelete='restrict')
+    src_dest_country_id = fields.Many2one(
+        'res.country', string='Origin/Destination Country',
+        ondelete='restrict')
+    intrastat_country = fields.Boolean(
+        related='src_dest_country_id.intrastat',
+        store=True, string='Intrastat Country', readonly=True)
+    intrastat = fields.Char(
+        string='Intrastat Declaration',
+        related='company_id.intrastat', store=True, readonly=True)
+
+    @api.multi
+    def onchange_partner_id(
+            self, type, partner_id, date_invoice=False,
+            payment_term=False, partner_bank_id=False, company_id=False):
+        res = super(AccountInvoice, self).onchange_partner_id(
+            type, partner_id, date_invoice=date_invoice,
+            payment_term=payment_term, partner_bank_id=partner_bank_id,
+            company_id=company_id)
+        if partner_id:
+            partner = self.env['res.partner'].browse(partner_id)
+            res['value']['src_dest_country_id'] = partner.country_id.id
+        return res
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = 'account.invoice.line'
+
+    hs_code_id = fields.Many2one(
+        'hs.code', string='Intrastat Code', ondelete='restrict')
+
+    @api.multi
+    def product_id_change(
+            self, product, uom_id, qty=0, name='', type='out_invoice',
+            partner_id=False, fposition_id=False, price_unit=False,
+            currency_id=False, company_id=None):
+        res = super(AccountInvoiceLine, self).product_id_change(
+            product, uom_id, qty=qty, name=name, type=type,
+            partner_id=partner_id, fposition_id=fposition_id,
+            price_unit=price_unit, currency_id=currency_id,
+            company_id=company_id)
+
+        if product:
+            product = self.env['product.product'].browse(product)
+            hs_code = product.get_hs_code_recursively()
```

Comment:
@damdam-s 
I am working on an update which already includes the fix suggested by you as well as a lot of other fixes and improvements.
I have done this work about a month ago but didn't submit a PR yet since there are still some pieces that I needed to complete and I have been busy on other projects.
I expect to finish this work (including the Belgian intrastat localisation on top of it) within the coming two days after which I'll make a PR. If you want to get a copy of what I currently have I can send it to you (but since I am working on it right now, the code is changing every minute and  hence I think it's better that you wait on the upcoming PR).